src/systemtest/resources/repo-config.csv

```diff
@@ -4,3 +4,4 @@ https://github.com/reposense/testrepo-Beta.git,master,fxml,docs**,yes,
 https://github.com/reposense/testrepo-Charlie.git,master,,,,9859b492760ca3a04432fcecc3ad64302a00689f;95aec8300a020c775d272d4f230d4ff5409793ef;b4b671e671ad3d917328322884c0e643b5b4d47d;80b2c05fcd8145df4301d6e57f89374c5e07cd28
 https://github.com/reposense/testrepo-Delta.git,master,,,,
```

Comment:
I've modified the implementation slightly as updating display name immediately after downloading the repository in GitDownloader caused the repoRoot to also get updated, causing errors subsequently as the repoRoot was not found. As you mentioned previously that ```repos``` directory can contain a constant or default such as HEAD, I've modified my implementation to ensure repoRoot for ```repos``` contains the default HEAD, while the displayName and branch has been updated and displayed accordingly in the report. I've therefore only removed the default for one of the branches in systemtest for now, if you are ok with this implementation i can proceed to remove the rest of the defaults.

Reply:
I would rather not have the `defaultDisplayName` because:
1. it doesn't serve any useful purpose.
2. its an additional field to maintain.

By repos directory can use a constant name, I mean that it can be named literally anything, even `TEMP` is fine, it doesn't have to use any unique identifier since its going to get deleted right after the analysis.

Moreover, even if you use a unique identifier, you could've just use a temp variable to store the run-time value instead of storing it as a class variable.